packages/vuetify/src/components/VSelect/VSelect.ts

```diff
@@ -550,10 +550,16 @@ export default baseMixins.extend<options>().extend({
       }
       this.selectedIndex = -1
     },
-    onClick () {
+    onClick (e: MouseEvent) {
       if (this.isDisabled) return
+      const appendInner = this.$refs['append-inner']
 
-      this.isMenuActive = true
+      if (!(appendInner &&
```

Comment:
Is it possible to make this more concise? It is hard to follow.

Reply:
I am unsure how to make this more concise, Would an added comment suffice? This is the same logic used at line 668 just flipped since we do not want to reopen the menu after onMouseUp would close it.

On thought I had would be to move this logic into its own function

```typescript
isAppendInner(e: MouseEvent) {
    // return true if append inner is present
    // and the target is itself or inside
    const appendInner = this.$refs['append-inner']
    return appendInner && (appendInner === e.target || (appendInner as { [key: string]: any }).contains(e.target))
}
```
then here it would look like 
```javascript
if (!this.isAppendInner(e)) {
```

Reply:
Let's go with that option. Just make sure to have a blank line between the variable declaration and the return statement.